plugins/discovery-file/build.gradle

```diff
@@ -27,3 +31,36 @@ bundlePlugin {
         into 'config'
     }
 }
+
+task seedTask << { }
+// setup the initial cluster with one node that will serve as the seed node
+// for unicast discovery
+ClusterConfiguration config = new ClusterConfiguration(project)
+config.distribution = 'zip'
+config.clusterName = 'discovery-file-test-cluster'
+List<NodeInfo> nodes = ClusterFormationTasks.setup(project, seedTask, config)
+
+// write the unicast_hosts.txt file to a temporary location to be used by the second cluster
+task setupSeedNodeAndUnicastHostsFile(type: DefaultTask) {
+    mustRunAfter(precommit)
+    dependsOn(seedTask)
+    doLast {
+        // write the unicast_hosts.txt file to a temp file in the build directory
+        File unicastHostsFile = file('build/unicast_hosts.txt')
+        unicastHostsFile.setText(nodes.get(0).transportUri(), 'UTF-8')
+    }
+}
+
+// second cluster, which will connect to the first via the unicast_hosts.txt file
+integTest {
+    dependsOn(setupSeedNodeAndUnicastHostsFile)
+    cluster {
+        distribution = 'zip'
+        clusterName = 'discovery-file-test-cluster'
+        plugin ':plugins:discovery-file'
+        // set the unicast.hosts in elasticsearch.yml to null, to ensure we are using the file-based plugin
+        unicastTransportUri = { seedNode, node, ant -> null }
```

Comment:
This is a single node cluster, so no unicast host setting will be added. So I don't think this is needed?